spec/controllers/login_controller_spec.rb

```diff
@@ -0,0 +1,48 @@
+# typed: false
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+# NOTE: While controller specs are generally frowned upon, and we have largely
+#       leaned on request specs instead, testing session-related code was not
+#       possible in a request spec, and this is application biz logic worth
+#       testing.
+RSpec.describe LoginController do
```

Comment:
Can you convert this to a request spec instead?

Reply:
No. See comment above.

Reply:
All the maintainers say that we should not be writing new controller specs: https://github.com/rspec/rspec-rails/issues/1838#issuecomment-313410705  We can test session related code in integration test (e.g. feature specs)

Reply:
@jcoyne since this a huge PR and I'd like to get it in before anything else, can I look into replacing it w/ a feature spec in a subsequent PR?